subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/HtmlPageGenerator.java

```diff
@@ -202,20 +202,19 @@ protected void renderHeaderForSamples(String label) {
                         .end();
                 }
             }
-        }
-    }
 
-    protected List<? extends PerformanceTestExecution> filterForRequestedCommit(List<? extends PerformanceTestExecution> results) {
-        String commitId = Git.current().getCommitId();
-        if (commitId == null) {
-            return results;
-        }
-        for (PerformanceTestExecution execution : results) {
-            if (execution.getVcsCommits().contains(commitId)) {
-                return results;
+            Optional<DataSeries<Duration>> first = executionVersions.stream().filter(Objects::nonNull).findFirst();
+            Optional<DataSeries<Duration>> last = Lists.reverse(executionVersions).stream().filter(Objects::nonNull).findFirst();
+            if (first.isPresent() && last.isPresent() && first.get() != last.get()) {
```

Comment:
This is a little bit more functional doing the same:
```java
            first.flatMap(firstData -> last.<Consumer<MetricsHtml>>map(lastData -> html -> {
                html.td().classAttr("numeric").text(FormatSupport.formatConfidence(firstData, lastData)).end();

                String differenceCss = FormatSupport.getDifference(first.get(), last.get()).doubleValue() > 0 ? "max-value" : "min-value";
                html.td().classAttr("numeric " + differenceCss).text(FormatSupport.formatDifference(first.get(), last.get())).end();
            })).orElseGet(() -> html -> {
                td().text("").end();
                td().text("").end();
            }).accept(this);
        }
```

Reply:
Isn't this... too functional? TBH this is a little complicated for me to understand with nested functions. I would prefer less functional way as it is right now.

Reply:
I am fine with leaving it as is.